Rewind - Signup flow: don't say that backups are ready if user skips the steps

eliorivero opened the pull request:
Fixes #22293

This PR fixes the issue where the text saying that backups were ready and backing up the site was shown **even if the user skipped the setup steps**.

With this PR:
- the text is no longer shown when user skips the steps. Applies to `/start/pressable-nux` and `/start/rewind-auto-config` flows
- the button in the last step for the `/start/pressable-nux` is now a proper primary button

### Step when backups were not configured

<img width="653" alt="captura de pantalla 2018-02-13 a la s 18 18 58" src="https://user-images.githubusercontent.com/1041600/36175944-d84989d8-10ef-11e8-80e3-c10943242a70.png">

### Step when backups are configured

<img width="652" alt="captura de pantalla 2018-02-13 a la s 18 19 14" src="https://user-images.githubusercontent.com/1041600/36175956-e51e3852-10ef-11e8-8e48-4e95ec0dcece.png">

### Testing

Please test with a Pressable site and a non-Pressable site
- site that doesn't have Rewind active because of the reason `vp_can_transfer`
- site in `awaiting_credentials` mode that requires the creds to be enter with the form
- site that can be auto-configured

If you need to re-test, make sure you clean Calypso's Local Storage and Indexed DB
<img width="244" alt="captura de pantalla 2018-02-13 a la s 18 20 31" src="https://user-images.githubusercontent.com/1041600/36176264-cabb760e-10f0-11e8-81b6-41310286fe04.png">

MichaelArestad commented on client/signup/steps/creds-complete/index.jsx:
```diff
@@ -21,25 +22,30 @@ class CredsCompleteStep extends Component {
 					{ translate(
-						'Your site is backing up because your site is set up with Jetpack Premium plan at ' +
-							'no additional cost to you. Finish setting up Jetpack and your site is ready to be ' +
+						'Finish setting up Jetpack and your site is ready to be ' +
```
Needs a space before `Finish`

rcoll replied:
Yeah this copy feels awkward to me as well.

MichaelArestad commented on client/signup/steps/creds-complete/index.jsx:
```diff
@@ -21,25 +22,30 @@ class CredsCompleteStep extends Component {
 
 		return (
 			<Card className="creds-complete__card">
 				<h3 className="creds-complete__title">{ translate( 'Your site is set up and ready!' ) }</h3>
```
This headline is a little weird in the context of adding credentials from the activity log. Would it be easy to change for just that flow? Perhaps the person could just exit the flow from the second skip link instead?

eliorivero replied:
@MichaelArestad yes, it's possible, and exiting through the second link would've required less refactoring. Please confirm if we'll go with that and I'll adapt things.

MichaelArestad replied:
Let's exit through the second skip link for the JITM/Banner flow.

rcoll commented:
On this page, after following the JITM to set up credentials, and skipping twice, I get the following heading:

![screen shot 2018-02-13 at 5 14 06 pm](https://user-images.githubusercontent.com/5528445/36176665-5be11914-10e1-11e8-95c4-7ff2a5f9d276.png)

beaulebens commented on client/signup/steps/rewind-were-backing/index.jsx:
```diff
@@ -25,21 +26,27 @@ class RewindWereBacking extends Component {
 					{ translate(
-						'Welcome to your Jetpack Premium plan! ' +
-							"That sound you're hearing? It's your website doing backflips with excitement!"
+						'Finish setting up Jetpack and your site is ready to be ' +
```
This sentence ends up butting up against the previous one -- needs a space in there.

eliorivero replied:
Good catch, thanks!

beaulebens commented on client/signup/steps/rewind-were-backing/index.jsx:
```diff
@@ -25,21 +26,27 @@ class RewindWereBacking extends Component {
 				<p className="rewind-were-backing__description rewind-switch__description">
+					{ get( dependencyStore, 'rewindconfig', false ) &&
+						translate(
+							'Your site is backing up because your site is set up with Jetpack Premium plan at ' +
```
This wording is pretty awkward, suggest:

"Your site is being backed up because it is set up with Jetpack Premium at no additional cost to you."

beaulebens replied:
(same above)

eliorivero replied:
Done 👍

rcoll commented on client/signup/config/steps.js:
```diff
@@ -262,12 +262,12 @@ export default {
 	'creds-confirm': {
 		stepName: 'creds-confirm',
-		providesDependencies: [],
+		providesDependencies: [ 'rewindconfig' ],
```
I'm curious why this is necessary here.

eliorivero replied:
@rcoll it's to whitelist the dependency, otherwise JS console throws a warning.

rcoll commented on client/signup/steps/rewind-were-backing/index.jsx:
```diff
@@ -25,21 +26,27 @@ class RewindWereBacking extends Component {
 					{ translate(
-						'Welcome to your Jetpack Premium plan! ' +
-							"That sound you're hearing? It's your website doing backflips with excitement!"
+						'Finish setting up Jetpack and your site is ready to be ' +
```
It could just be me, but this copy feels awkward and unclear.

MichaelArestad replied:
Yeah, I thought it odd as well. We might want a specific version for folks setting up creds via the JITM?

rcoll commented on client/signup/steps/rewind-were-backing/index.jsx:
```diff
@@ -25,21 +26,27 @@ class RewindWereBacking extends Component {
 				</h3>
 				<img src="/calypso/images/illustrations/thankYou.svg" alt="" />
 				<p className="rewind-were-backing__description rewind-switch__description">
+					{ get( dependencyStore, 'rewindconfig', false ) &&
```
This does work, but it seems like we should be checking the SM here, rather than relying on the dependency store.

eliorivero replied:
Makes sense 👍
I'll wait until Michael responds about exiting the flow in the second skip.

eliorivero commented:
@rcoll that screenshot with the "We're backing up your site" is odd because that string was actually removed in https://github.com/Automattic/wp-calypso/pull/22411/files#diff-38269bd48feacb757467e037ffb3626dL36

rcoll commented:
@eliorivero I got sent to Production when I clicked the JITM on my first test. Just tested again locally and didn't get the message. False alarm, looks fine!

eliorivero commented:
Thanks for the review, I've addressed the issues:

- in the /start/rewind-auto-config flow, go back to AL if user skips a second time
- added space to separate sentences
- updated "Your site is being backed up..." wording

MichaelArestad commented:
Nice. Ship it!
